client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/NodeBuilder.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.client.opentracing;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.hawkular.apm.api.model.Property;
+import org.hawkular.apm.api.model.trace.Component;
+import org.hawkular.apm.api.model.trace.Consumer;
+import org.hawkular.apm.api.model.trace.ContainerNode;
+import org.hawkular.apm.api.model.trace.CorrelationIdentifier;
+import org.hawkular.apm.api.model.trace.Node;
+import org.hawkular.apm.api.model.trace.NodeType;
+import org.hawkular.apm.api.model.trace.Producer;
+
+/**
+ * This class is responsible for building up the information related to a particular node
+ * within a trace fragment. The actual node type may not be known until part way through
+ * this process. So instead a tree of node builders is constructed, and only converted into
+ * a node tree as a final step.
+ *
+ * @author gbrown
+ */
+public class NodeBuilder {
```

Comment:
The Node class being built is in the api module - its possible we could make this more generic once happy with the approach.